controller/orphan_controller.go

```diff
@@ -386,16 +403,16 @@ func (oc *OrphanController) cleanupOrphanedReplicaInstance(orphan *longhorn.Orph
 		return false, err
 	}
 
-	var spec *longhorn.InstanceSpec
+	var status *longhorn.InstanceStatus
 	if replicaCR, err := oc.ds.GetReplicaRO(instance); err != nil {
 		if !datastore.ErrorIsNotFound(err) {
 			return false, err
 		}
-		spec = nil
+		status = nil
 	} else {
-		spec = &replicaCR.Spec.InstanceSpec
+		status = &replicaCR.Status.InstanceStatus
 	}
-	oc.cleanupOrphanedInstance(orphan, instance, imName, longhorn.InstanceManagerTypeReplica, spec)
+	oc.cleanupOrphanedInstance(orphan, instance, imName, longhorn.InstanceManagerTypeReplica, status)
```

Comment:
What's the purpose of the change?

Reply:
Inside `cleanupOrphanedInstance`, we change the instance ownership check logic, compares the `Status.InstanceManagerName` instead of `Spec.NodeID`.

Reply:
When the `instance.Status.InstanceManagerName` is the same with the one inside orphan CR, it means the instance is scheduled back to the instance manager, the instance process is no longer considered orphaned, and the engine/replica controller will manage this process. The orphan controller should finalize the orphan CR without touching the instance process.